### PR TITLE
Migrate system dataset logic to middlewared

### PR DESCRIPTION
**Pull request by william-gr**

### CODE REVIEW

**Comment by william-gr** [comment]
Jenkins: retest this please

**Comment by kmoore134** [comment]
Jenkins: retest this please

File: src/middlewared/middlewared/plugins/sysdataset.py [hunk 1]
```diff
@@ -1,9 +1,296 @@
...
+        return await self.middleware.call('datastore.config', 'system.systemdataset', {'prefix': 'sys_', 'extend': 'systemdataset.config_extend'})
+
+    @private
+    async def config_extend(self, config):
+
+        # Add `is_decrypted` dynamic attribute
+        if config['pool'] == 'freenas-boot':
+            config['is_decrypted'] = True
+        else:
+            pool = await self.middleware.call('pool.query', [('name', '=', config['pool'])])
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/sysdataset.py]: Maybe we should add `query_first` and `query_one` methods like these:
http://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.first
http://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.one

Comment (william-gr) [reply src/middlewared/middlewared/plugins/sysdataset.py]: We kind of already have the "one"
self.middleware.call('pool.query', [('name', '=', config['pool'])], {'get': True})

However it raises an exception if not found. Perhaps it should be changed to return None

File: gui/middleware/notifier.py [hunk 2]
```diff
@@ -884,7 +883,8 @@ def list_zfs_fsvols(self, system=False):
         out = out.split('\n')
         retval = OrderedDict()
         if system is False:
```
[the remark this answers is not in the document]
Comment (william-gr) [reply gui/middleware/notifier.py]: Hm ok, but not sure I understand, what changes moving the if condition to the beginning?

**Pull request merged**